cmd/global-heal.go

```diff
@@ -98,19 +98,42 @@ func healErasureSet(ctx context.Context, setIndex int, xlObj *xlObjects) error {
 	// Heal all buckets with all objects
 	for _, bucket := range buckets {
 		// Heal current bucket
-		bgSeq.sourceCh <- bucket.Name
+		bgSeq.sourceCh <- healSource{
+			path: bucket.Name,
+		}
 
 		// List all objects in the current bucket and heal them
 		listDir := listDirFactory(ctx, xlObj.getLoadBalancedDisks()...)
 		walkResultCh := startTreeWalk(ctx, bucket.Name, "", "", true, listDir, nil)
 		for walkEntry := range walkResultCh {
-			bgSeq.sourceCh <- pathJoin(bucket.Name, walkEntry.entry)
+			bgSeq.sourceCh <- healSource{
+				path: pathJoin(bucket.Name, walkEntry.entry),
+			}
 		}
 	}
 
 	return nil
 }
 
+// deepHealObject heals given object path in deep to fix bitrot.
+func deepHealObject(ctx context.Context, objectPath string) {
+	// Get background heal sequence to send elements to heal
+	var bgSeq *healSequence
+	var ok bool
+	for {
+		bgSeq, ok = globalBackgroundHealState.getHealSequenceByToken(bgHealingUUID)
+		if ok {
+			break
+		}
+		time.Sleep(time.Second)
+	}
+
+	bgSeq.sourceCh <- healSource{
```

Comment:
Will this block until it is picked up?

Maybe add a size to the channel and do:

```
select {
case bgSeq.sourceCh <- healSource{....}:
default:
}
```

We don't want reads blocking to kick off healing.

Reply:
same as above

Reply:
Yep. Channel write blocks till its written. Instead of buffered channel, I can call this function in a goroutine.